src/main/java/org/jenkinsci/plugins/docker/commons/tools/DockerToolInstaller.java

```diff
@@ -142,6 +146,27 @@ public FilePath performInstallation(ToolInstallation toolInstallation, @Nonnull
         return install;
     }
 
+    static URL getDockerImageUrl(String os, String version) throws MalformedURLException {
+        if (parseVersion(version).isNewerThan(parseVersion("17.05.0-ce")))
+            return new URL("https://download.docker.com/" + os + "/docker-" + version);
+
+        String osName="";
+        if (os.startsWith("linux")) osName = "Linux";
+        if (os.startsWith("win")) osName = "Windows";
+        if (os.startsWith("mac")) osName = "Darwin";
+        return new URL("https://get.docker.com/builds/" + osName + os.substring(os.lastIndexOf("/")) + "/docker-" + version);
+    }
+
+    private static VersionNumber parseVersion(String version) {
+        // any version that sorts before 17.05.0-ce
+        if (version.equals("latest")) return new VersionNumber("0");
```

Comment:
I would rather prefer an API with `null` and annotations, but fine for a private method